src/index.ts

```diff
@@ -25,145 +88,275 @@ export const installCondaPackage = async (
         }
       });
 
-      if (condaPackage === undefined || packageInfo === undefined) {
+      if (
+        (condaPackage && condaPackage.byteLength === 0) ||
+        (packageInfo && packageInfo.byteLength === 0)
+      ) {
         throw new Error(`Invalid .conda package ${url}`);
       }
-
       const condaFiles: FilesData = await untarjs.extractData(condaPackage);
-      const packageInfoFiles: FilesData = await untarjs.extractData(packageInfo);
-
-      createCondaMetaFile(packageInfoFiles, prefix, FS, verbose);
-      saveFiles(prefix, FS, {...condaFiles, ...packageInfoFiles}, verbose);
+      const packageInfoFiles: FilesData =
+      await untarjs.extractData(packageInfo);
+      saveCondaMetaFile(packageInfoFiles, newPrefix, FS, verbose);
+      saveFiles(FS, { ...condaFiles, ...packageInfoFiles }, newPrefix);
+      return condaFiles;
     } else {
-      createCondaMetaFile(files, prefix, FS, verbose);
-      saveFiles(prefix, FS, files, verbose);
+      saveCondaMetaFile(files, newPrefix, FS, verbose);
+      saveFiles(FS, files, newPrefix);
+      return files;
     }
-
-    return;
   }
 
   throw new Error(`There is no file in ${url}`);
 };
+const getSharedLibs = (files: FilesData, prefix: string): FilesData => {
+  let sharedLibs: FilesData = {};
 
-const saveFiles = (
-  prefix: string,
-  FS: any,
-  files: FilesData,
-  verbose: boolean
-): void => {
-  console.log('Saving files into browser memory');
+  Object.keys(files).map(file => {
+    if (file.endsWith('.so') || file.includes('.so.')) {
+      sharedLibs[`${prefix}/${file}`] = files[file];
+    }
+  });
+  return sharedLibs;
+};
+
+function getParentDirectory(filePath: string) {
+  return filePath.substring(0, filePath.lastIndexOf('/'));
+}
 
+const saveFiles = (FS: any, files: FilesData, prefix: string): void => {
   try {
-    ['site-packages', 'info', 'etc', 'share'].forEach(folder => {
-      let folderDest = `${prefix}/${folder}`;
-      if (folder === 'site-packages') {
-        folderDest = `${prefix}/lib/python3.11/site-packages`;
+    Object.keys(files).forEach(filename => {
+      const dir = getParentDirectory(filename);
+      if (!FS.analyzePath(dir).exists) {
+        FS.mkdirTree(dir);
       }
-      savingFiles(files, folder, folderDest, FS, verbose);
+
+      let encodedData = new TextDecoder('utf-8').decode(files[filename]);
+      FS.writeFile(`${prefix}/${filename}`, encodedData);
```

Comment:
This is the culprit! 

```suggestion
      FS.writeFile(`${prefix}/${filename}`, files[filename]);
```

Reply:
done